text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1742 @@
+the table.
+
+```typescript
+  const globalTable = new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    sortKey: { name: 'Bar', type: AttributeType.STRING },
+    localSecondaryIndex: [
+      {
+        indexName: 'FooTableLsi',
+        sortKey: { name: 'Foo', type: AttributeType.STRING },
+      },
+    ],
+  });
+```
+
+You can add local secondary index with `addLocalSecondaryIndex` method.
+
+```typescript
+  globalTable.addLocalSecondaryIndex({
+    indexName: 'FooTableLsi',
+    sortKey: { name: 'Foo', type: AttributeType.STRING },
+  });
+```
+
+__NOTE:__
+
+* You need a sort key to define a local secondary index.
+* You can create up to five local secondary indexes.
+
+#### Encryption
+
+There are three types of encryptions that are available for global tables.
+
+1. `AWS_OWNED`: This uses a KMS key for encryption that is owned by DynamoDB. This is the default
+for global tables.
+2. `AWS_MANAGED`: A KMS key is created in your account and is managed by AWS.
+3. `CUSTOMER_MANAGED`: You can specify a KMS key that is managed by you for encrytion.
+
+This would be the same for each replica.
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    encryption: TableEncryption.AWS_MANAGED,
+  });
+```
+
+__NOTE__:
+
+* If the encryption mode is CUSTOMER_MANAGED, then you would need to specify a KMS key for each replica.
+
+#### Streams
+
+You can enable DynamoDB streams for your global tables. There are four possible values for it,
+
+1. `NEW_IMAGE`: The entire item, as it appears after it was modified, is written to the stream.
+2. `OLD_IMAGE`: The entire item, as it appeared before it was modified, is written to the stream.
+3. `NEW_AND_OLD_IMAGES`: Both the new and the old item images of the item are written to the stream.
+4. `KEYS_ONLY`: Only the key attributes of the modified item are written to the stream.
+
+This would be the same for each replica. By default, streams are not enabled for global table. But, if
+you have a replica present which is not of the same region as where the stack is deployed to, then the
+stream type would be set to `NEW_AND_OLD_IMAGES` automatically since it is required for table replication.
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    stream: StreamViewType.NEW_IMAGE,
+  });
+```
+
+#### Time To Live
+
+You can enable time to live for an attribute of the global table. This would be the same for each replica.
+By default, it is not enabled for any attributes.
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    timeToLiveAttribute: 'Foo',
+  });
+```
+
+#### Replicas
+
+You can define replicas for your global table. By default, if this configuration is undefined, then the
+table is deployed to the region where stack is being deployed.
```

Comment:
This sentence is not clear to me. If my `replicas` array has one element, does that lead to a `table + replica`, or just a `table` (replacing all the properties at the top level) ?

I think I'd prefer the first behavior?